packages/api/models/schemas.js

```diff
@@ -295,7 +295,6 @@ module.exports.granule = {
     },
     published: {
       type: 'boolean',
-      default: false,
       description: 'shows whether the granule is published to CMR',
       readonly: true,
     },
```

Comment:
So I guess there aren't problems here if `published: undefined` even though that violates the type constraint. Is that because we'd need to call `recordIsValid` without `published` set to true/false and we've corrected the instances where that happens?

Reply:
> So I guess there aren't problems here if published: undefined even though that violates the type constraint. 

off-hand, I believe in json boolean can be true/false/null.   Undefined should be parsed to 'missing key'. 

A test in the granule validation units, as an example, passes according to the comments below: 


```
test.only('storeGranule() fails validation if granule published is <value>', async (t) => {
  const {
    granuleModel,
    collectionId,
    provider,
    workflowStatus,
  } = t.context;

  const bucket = randomId('bucket');
  await S3.createBucket(bucket);
  t.teardown(() => S3.recursivelyDeleteS3Bucket(bucket));

  const granule1 = fakeGranuleFactoryV2({
    files: [fakeFileFactory({ bucket })],
  });

  await S3.s3PutObject({ Bucket: bucket, Key: granule1.files[0].key, Body: 'asdf' });

  const files = await granuleModel.fileUtils.buildDatabaseFiles({
    s3: awsClients.s3(),
    providerURL: buildURL(provider),
    files: granule1.files,
  });

  const granuleRecord = await generateGranuleApiRecord({
    granule: granule1,
    files,
    executionUrl: 'http://execution-url.com',
    collectionId,
    provider: provider.id,
    workflowStatus,
    status: getGranuleStatus(workflowStatus, granule1),
    cmrUtils,
    updatedAt: Date.now(),
  });

  // delete granuleRecord.published; //-- pass
  // granuleRecord.published = false; -- pass
  // granuleRecord.published = undefined; -- pass
  await granuleModel.storeGranule(granuleRecord);

  t.true(await granuleModel.exists({ granuleId: granule1.granuleId }));
});
```

Reply:
Note - ^ was off, null fails.

Reply:
Updated above to address the nullish delete to false concern.